### ISSUE TITLE
get_documents

### ISSUE THREAD

**Issue by mzkhuzam2** [judgelord/regulationsdotgov#14]
- not sure if it needs a batch option, will a docket have more than 5k docs? 
- not currently working in sandbox, working on it

**Comment by judgelord** [comment]
I don't think we would end up with more than 5k. 

Before seeing this. I made some edits to this go get it working, but it could be cleaner without the batch argument

On thing to keep in mind: the api_key or api_keys argument cannot be used as the default in more than one place. It is ok if the top-level function defaults to api_keys = api_keys, but we can't then do that in helper functions or it gets recursive: https://stackoverflow.com/questions/4357101/promise-already-under-evaluation-recursive-default-argument-reference-or-earlie

**Comment by judgelord** [comment]
I did get one that was a bit over 3000 documents, but have not seen 5k

**Comment by judgelord** [comment]
I'd say we can drop the batching for now, and put it back in if it is needed.

**Comment by judgelord** [comment]
investigate: EPA-HQ-OW-2009-0819

**Comment by judgelord** [comment]
😳EPA-HQ-OW-2009-0819 has 7,196 documents @mzkhuzam2

**Comment by mzkhuzam2** [comment]
This is a really weird docket. It has a Unified Agenda tab which I haven't seen before, documents have been posted from 2013-2023, and it's received 300k+ comments but only 4k have been posted.

Should we hold off on dropping batching for now?

**Comment by judgelord** [comment]
yes, it will be more robust to odd things like this if it has the batch loop, so let's keep it.

**Comment by mzkhuzam2** [comment]
> I don't think we would end up with more than 5k.
> 
> Before seeing this. I made some edits to this go get it working, but it could be cleaner without the batch argument
> 
> On thing to keep in mind: the api_key or api_keys argument cannot be used as the default in more than one place. It is ok if the top-level function defaults to api_keys = api_keys, but we can't then do that in helper functions or it gets recursive: https://stackoverflow.com/questions/4357101/promise-already-under-evaluation-recursive-default-argument-reference-or-earlie

Okay, I think I established a system that works: 
**Two types of API key calls: api_keys, api_key** 

- api_key is reserved for the make_call function because that will loop through the different keys 
- api_keys is in the function arguments for both the batch and higher-level functions, but it is not set with a default value anywhere. Users must set their own default value. 

It seems in the StackOverflow responses, the issue arises from setting a default argument and not the naming standard across the different functions so I think this makes sense. Let me know if you run into any issues. 

Note: I'm not sure where "keys" is used .... I know I saw it somewhere but I can't find it anymore. If that causes a problem let me know.

**Comment by mzkhuzam** [comment]
I investigated DOT-OST-2005-23307 today, and it looks like the majority of the documents are actually comments. I tried to collect the comments on all 5084 document to see what would be returned, but it kept failing and I got frustrated. I'll try again later once my API hit rate refreshes.
